style(editors): prefer `const` to `let`

overlookmotel opened the pull request:
Use `const` where variables are not reassigned instead of `let`. Preparation for enabling ESLint's `prefer-const` rule (running as an Oxlint JS plugin) in our linting setup.

overlookmotel commented:
I think this is uncontroversial, but `editors` is not my area, so I don't want to merge myself.

camc314 commented:
### Merge activity

* **Dec 2, 2:21 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 2, 2:25 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
style(editors): prefer `const` to `let` (#16395)

Use `const` where variables are not reassigned instead of `let`. Preparation for enabling ESLint's `prefer-const` rule (running as an Oxlint JS plugin) in our linting setup.

overlookmotel commented:
> > Preparation for enabling ESLint's prefer-const rule (running as an Oxlint JS plugin) in our linting setup.
> 
> Alternatively.... #15707

Oh great! I do like this rule!

However, part of the purpose is also to test out JS plugins. Once your Rust implementation is merged, I'll find another ESLint rule to use for this purpose.
